core/base-service/node-fetch.js

```diff
@@ -0,0 +1,101 @@
+import { URL, URLSearchParams } from 'url'
+import fetch from 'node-fetch'
+import { Inaccessible, InvalidResponse } from './errors.js'
+
+const userAgent = 'Shields.io/2003a'
+
+function object2URLSearchParams(obj) {
+  const qs = {}
+  for (const [key, value] of Object.entries(obj)) {
+    if (value === undefined) {
+      continue
+    } else if (value === null) {
+      qs[key] = ''
+    } else if (['string', 'number', 'boolean'].includes(typeof value)) {
+      qs[key] = value
+    }
+  }
+  return new URLSearchParams(qs)
+}
+
+function request2NodeFetch({ url, options }) {
```

Comment:
There's a bit of a separation of concerns issue going on here.
Part of what I am doing here is translating request concepts to node-fetch concepts
That said, if we go with node-fetch I think there are some abstractions here that we would probably want to retain as an abstraction available to services. For example the qs/form abstraction. I think for the purpose of evaluating performance this is fine and we can clean things up if we decide node-fetch is the way forward.